File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.cpp
class Solution {
public:
    int maxScore(string s) {
        int n=s.length();
        vector<int>pref(n,0);
        int cnt=0;
        for(int i=n-1;i>=0;i--){
            if(s[i]=='1') cnt++;
            pref[i]=cnt;
        }
        // for(int i: pref) cout<< i<<" ";
        // cout<<endl;
        int ans=0;
        cnt=0;
       // if(s[0]=='0' ) cnt++;
        for(int i=0;i<n-1;i++){
            
            if(s[i]=='0') cnt++;
            ans=max(ans,cnt+pref[i+1]);
            
        }
        return ans;
    }
};

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        one=0
        zero=0
        tone=0
        tzero=0
        ans=0
        for i in s:
            if i=='0':
                tzero+=1
            else:
                tone+=1
        
        for i in range(len(s)-1):
            if s[i]=='0':
                zero+=1
            else:
                one+=1
            ans=max(ans,zero+(tone-one))

        return ans